src/exeptions/api.error.js

```diff
@@ -0,0 +1,32 @@
+export class ApiError extends Error {
+  constructor({ message, status, errors = {} }) {
+    super(message);
+
+    this.status = status;
+    this.errors = errors;
+  }
+
+  static badRequest(message, errors) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    });
+  }
+
+  static unauthorize(errors) {
```

Comment:
The method name `unauthorize` might be misleading. Consider renaming it to `unauthorized` to better reflect its purpose and maintain consistency with HTTP status code terminology.